backend/src/routes/auth.routes.ts

```diff
@@ -1,18 +1,40 @@
+import { JWTRolePayload } from './../types';
+import { ROLES } from './../models/user.model';
 import { Router, Request, Response, NextFunction } from 'express';
 import AuthController from '../controllers/auth.controller';
 import jwt from 'jsonwebtoken';
 
 const router: Router = Router();
 const authController: any = new AuthController();
 
-const authJWT = (req: Request, res: Response, next: NextFunction): Response => {
+const jwtRolePayload = (token: string): JWTRolePayload => JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString());
+
+export const checkJWTExpired = (req: Request, res: Response, next: NextFunction): Response => {
     const authHeader: string = req.headers.authorization;
 
     if (authHeader) {
         const token: string = authHeader.split(' ')[1];
 
         jwt.verify(token, process.env.JWT_TOKEN_KEY, (err) => {
-            if (err) return res.status(401).json({auth: false});
+            if (err) return res.status(401).json('Token expired');
+
+            next();
+        });
+    } else {
+        return res.sendStatus(401);
+    }
+};
+export const checkOnlyAdmin = (req: Request, res: Response, next: NextFunction): Response => {
```

Comment:
Create something like checkRole(role: ROLES): boolean
Inside compare the role from JWT and role from args